libdevcrypto/Common.cpp

```diff
@@ -58,6 +58,9 @@ class Secp256k1Context
 
 bool dev::SignatureStruct::isValid() const noexcept
 {
+	if (zeroSign()) {
```

Comment:
Please be careful, this function is used inside the ecrecover precompile whose behaviour should not be changed.
I think it might be better to keep `isValid()` returning false and treat the zero-signature as a special case from the outside.